### PR TITLE
Adding COUNTER_MARGIN to Ingress drop counters for qos tests

### PR DESCRIPTION
**Pull request by ansrajpu-git**
### Description of PR

Adding COUNTER_MARGIN to ingress drop counter for below test scenarios to avoid flakey results.

- TestQosSai.testQosSaiPfcXoffLimit
- TestQosSai.testQosSaiPfcXonLimit 
- TestQosSai.testQosSaiLossyQueue

Summary:
Sometimes for dnx, few extra ipv6 NS/RA pkt received from VM which get added to counter value & may give inconsistent test results.
Adding COUNTER_MARGIN to provide room to 2 extra pkt incase, extra traffic received. This will avoid inconsistent results on qos test.
Below tests are updated:
- TestQosSai.testQosSaiPfcXoffLimit
- TestQosSai.testQosSaiPfcXonLimit 
- TestQosSai.testQosSaiLossyQueue
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

### Approach
Sometimes for dnx, few extra ipv6 NS/RA pkt received from VM which get added to counter value & may give inconsistent test results.
Providing room for 2 pkts (COUNTER_MARGIN) incase, extra traffic received. 
Below test are updated:
- TestQosSai.testQosSaiPfcXoffLimit
- TestQosSai.testQosSaiPfcXonLimit 
- TestQosSai.testQosSaiLossyQueue
- 
#### What is the motivation for this PR?
Inconsistent results on qos test runs
#### How did you verify/test it?
Executed qos tests & verify the results.

### CODE REVIEW

File: tests/saitests/py3/sai_qos_tests.py [hunk 1]
```diff
@@ -1393,18 +1414,20 @@ def runTest(self):
...
-            else:
-                # recv port ingress drop
-                for cntr in ingress_counters:
-                    assert (recv_counters[cntr] > recv_counters_base[cntr]
-                            ), 'unexpectedly RX drop counter not increase, {}'.format(test_stage)
-                # xmit port no egress drop
-                for cntr in egress_counters:
-                    assert (xmit_counters[cntr] == xmit_counters_base[cntr]
-                            ), 'unexpectedly TX drop counter increase, {}'.format(test_stage)
+            # For dnx few extra ipv6 NS/RA pkt received from VM, adding to counter value
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:1417]: @ansrajpu-git  Original code is stating these counters not supported for DNX. Is this not the case ?

Comment (saksarav-nokia) [reply tests/saitests/py3/sai_qos_tests.py:1417]: > @ansrajpu-git Original code is stating these counters not supported for DNX. Is this not the case ?

vineet, these drop counters were not supported initially and they are supported now. So enabling the check

File: tests/saitests/py3/sai_qos_tests.py [hunk 2]
```diff
@@ -3328,8 +3360,15 @@ def runTest(self):
...
             # recv port no ingress drop
+            # For dnx few extra ipv6 NS/RA pkt received, adding to coutner value
+            # & may give inconsistent test results
+            # Adding COUNTER_MARGIN to provide room to 2 pkt incase, extra traffic received
             for cntr in ingress_counters:
                 if platform_asic and platform_asic == "broadcom-dnx" and cntr == 1:
+                    print("recv_counters_base: %d, recv_counters: %d" % (recv_counters_base[cntr],
+                                                                         recv_counters[cntr]), file=sys.stderr)
+                    assert (recv_counters[cntr] <= recv_counters_base[cntr] + COUNTER_MARGIN)
+                else:
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:3371]: @saksarav-nokia @ansrajpu-git This seems to be added for non chassis. Is this intentional ?

**Review by judyjoseph — APPROVED** [review]

File: tests/saitests/py3/sai_qos_tests.py (hunk 2 above)
Comment (vmittal-msft) [reply tests/saitests/py3/sai_qos_tests.py:3370]: Looks like this is fixing mistake from previous commit. Looks good.

**Comment by judyjoseph** [comment]
@ansrajpu-git Could you raise a separate PR 202205 branch ? as there are conflicts

**Pull request merged**